### PR TITLE
Update doc about literals from ReST to Markdown

### PR DESCRIPTION
**Pull request by danielmartin**
#### What's in this pull request?

Convert the literals documentation from reStructuredText to Markdown.

/cc @jrose-apple

### CODE REVIEW

File: docs/Literals.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+The StringLiteralConvertible Protocol
+-------------------------------------
+
+Here is the StringLiteralConvertible protocol as defined in the
+standard library’s CompilerProtocols.swift:
+
+``` swift
+    // NOTE: the compiler has builtin knowledge of this protocol
+    // Conforming types can be initialized with arbitrary string
+    literals.  public protocol StringLiteralConvertible
```
Comment (mominul) [inline docs/Literals.md]: You might want to write `// literals`

Comment (danielmartin) [inline docs/Literals.md]: I've fixed some formatting issues with code.

**Comment by jrtibbetts** [comment]
👍

**Comment by jrose-apple** [comment]
Hm. We kind of stalled on the ReST-to-Markdown conversion. @tkremenek?

**Comment by tkremenek** [comment]
I agree we should complete the transition to Markdown.

@danielmartin Thanks for doing this!

**Pull request merged**